src/scales/scale.timeseries.js

```diff
@@ -1,29 +1,35 @@
 import TimeScale from './scale.time';
-import {_lookupByKey} from '../helpers/helpers.collection';
+import {_lookup} from '../helpers/helpers.collection';
 
 /**
- * Linearly interpolates the given source `value` using the table items `skey` values and
- * returns the associated `tkey` value. For example, interpolate(table, 'time', 42, 'pos')
- * returns the position for a timestamp equal to 42. If value is out of bounds, values at
- * index [0, 1] or [n - 1, n] are used for the interpolation.
+ * Linearly interpolates the given source `val` using the table. If value is out of bounds, values
+ * at index [0, 1] or [n - 1, n] are used for the interpolation.
  * @param {object} table
- * @param {string} skey
- * @param {number} sval
- * @param {string} tkey
+ * @param {number} val
+ * @param {boolean} [reverse]
  * @return {object}
  */
-function interpolate(table, skey, sval, tkey) {
-	const {lo, hi} = _lookupByKey(table, skey, sval);
+function interpolate(table, val, reverse) {
+	let prevSource, nextSource, prevTarget, nextTarget;
 
 	// Note: the lookup table ALWAYS contains at least 2 items (min and max)
-	const prev = table[lo];
-	const next = table[hi];
-
-	const span = next[skey] - prev[skey];
-	const ratio = span ? (sval - prev[skey]) / span : 0;
-	const offset = (next[tkey] - prev[tkey]) * ratio;
+	if (reverse) {
+		// going from position to time
```

Comment:
Needing an comment here suggests variable names are not good.
[prev/next][Timestamp/Index]?

Reply:
The difficulty is that the variables aren't consistently used as timestamp or index. They mean one thing in the normal case and the opposite in the `reverse` case, so I'm not sure we'll be able to do better than source/target. I moved the comments to the JSDoc description of `reverse` instead